src/chunk_index.c

```diff
@@ -925,6 +925,55 @@ chunk_index_mark_clustered(Oid chunkrelid, Oid indexrelid)
 	heap_close(rel, AccessShareLock);
 }
 
+/*
+ * Create versions of every index over src_chunkrelid over chunkrelid.
+ * Returns the relids of the new indexes created.
+ * New indexes are in the same order as RelationGetIndexList.
+ */
+List *
+chunk_index_duplicate(Oid src_chunkrelid, Oid dest_chunkrelid)
+{
+	Relation	hypertable_rel = NULL;
+	Relation	src_chunk_rel;
+	Relation	dest_chunk_rel;
+	List	   *index_oids;
+	Oid			constraint_oid;
+	ListCell   *index_elem;
+	List	   *new_index_oids = NIL;
+	Chunk	   *src_chunk;
+
+	/* TODO lock order? */
+	src_chunk_rel = relation_open(src_chunkrelid, AccessShareLock);
```

Comment:
would be nice to keep all the opens together. i.e. move line 949 up. The order of open and close should be reversed. also why mix relation_open and heap open?

Reply:
Not sure why I made one a `relation_open`, changed it to `heap` like the others.
I'd like to keep all the `open`s together, but I need o get `src_chunk` from `src_chunk_rel` before I know which hypertable to open, so I think there will have to be a little intermixing.